tests/unix_stream.rs

```diff
@@ -444,6 +462,14 @@ where
     expect_read!(stream.read(&mut buf), DATA1);
 
     assert!(stream.take_error().unwrap().is_none());
+    // TODO: have to re-register here to reset user_events
```

Comment:
Is this expected behavior on windows given this note in the selector? If so, I can make this re-registration in the test conditional on the platform

https://github.com/tokio-rs/mio/blob/c6b5f13adf67483d927b1766faf984c1871e8abc/src/sys/windows/selector.rs#L230-L233

Reply:
The tests should work the same across platforms, so this needs to change in the implementation.

Reply:
Given https://github.com/tokio-rs/mio/issues/1611 I think the only reason these tests are failing is that there is not a non-blocking read call after the initial read, as there is in the corresponding `tcp` tests 🤦 

I think adding a non-blocking read here could fix these tests for now, rather than changing draining behavior.